ouroboros-network/src/Ouroboros/Network/PeerSelection/Churn.hs

```diff
@@ -124,17 +162,26 @@ peerChurnGovernor tracer churnTracer
       -- ^ counter getter
       -> DiffTime
       -- ^ timeout
-      -> ModifyPeerSelectionTargets
+      -> (ChurnRegime -> HotValency -> PeerSelectionTargets -> ModifyPeerSelectionTargets)
       -- ^ update counters function
       -> CheckPeerSelectionCounters
       -- ^ check counters
       -> m ()
     updateTargets churnAction getCounter timeoutDelay modifyTargets checkCounters = do
       -- update targets, and return the new targets
       startTime <- getMonotonicTime
-      (c, targets) <- atomically $
+      (c, targets) <- atomically $ do
+        churnMode <- updateChurnMode
+        ltt       <- getLocalRootHotTarget
+        regime <- pickChurnRegime consensusMode churnMode <$> getUseBootstrapPeers
+        let base =
+              case (consensusMode, churnMode) of
+                (GenesisMode, ChurnModeBulkSync) -> genesisSyncTargets
+                (GenesisMode, ChurnModeNormal)   -> praosTargets
+                (PraosMode, _)                   -> praosTargets
```

Comment:
Shouldn't we use `LedgerStateJudgement` here instead of `ChurnMode`.  This will be mroe consistent with `Monitoring.targets` actions.

Reply:
The pure logic should be put in a function
```
getPeerSelectionTargets :: ConsensusMode -> LedgerStateJudgement -> ConensusModePeerTargets -> PeerSelectionTargets
```
and then used here and in the `Monitor.targets` where we detect if churn is going on.  This way, we'll ensure these parts are doing the same thing.

Reply:
LedgerPeersConsensusInterface is not passed in to churn - it's no problem I can add it, but I thought that FetchMode is synchronized with ledger state judgement by @bolt12's patch, no?

Reply:
Yes, that's correct, when in TooOld state we are guaranteed to be in BulkSync mode, however the contrary is not guaranteed! So I would say it is indeed better to use ledger state judgement if that's the source of truth we care about for this piece of logic.

Reply:
Ok, thanks for the note, I'll take a closer look and update the code accordingly.